src/protagonist/CleanupHandler/AssetUpdatedHandler.cs

```diff
@@ -0,0 +1,357 @@
+                break;
+            case AssetDeliveryChannels.Timebased:
+                await CleanupRemovedTimebasedDeliveryChannel(assetAfter);
+                break;
+            case AssetDeliveryChannels.File:
+                CleanupFileDeliveryChannel(assetAfter);
+                break;
+            default:
+                logger.LogDebug("policy {PolicyName} does not require any changes for asset {AssetId}",
+                    deliveryChannelRemoved.DeliveryChannelPolicy.Name, assetAfter.Id);
+                break;
+        }
+    }
+
+    private async Task CleanupChangedPolicy(ImageDeliveryChannel deliveryChannelModified, Asset assetAfter)
+    {
+        switch (deliveryChannelModified.Channel)
+        {
+            case AssetDeliveryChannels.Image:
+                CleanupChangedImageDeliveryChannel(deliveryChannelModified, assetAfter.Id);
+                break;
+            case AssetDeliveryChannels.Thumbnails:
+                await CleanupChangedThumbnailDeliveryChannel(assetAfter);
+                break;
+            case AssetDeliveryChannels.Timebased:
+                await CleanupChangedTimebasedDeliveryChannel(deliveryChannelModified, assetAfter.Id);
+                break;
+            default:
+                logger.LogDebug("policy {PolicyName} does not require any changes for asset {AssetId}",
+                    deliveryChannelModified.DeliveryChannelPolicy.Name, assetAfter.Id);
+                break;
+        }
+    }
+
+    private async Task CleanupChangedTimebasedDeliveryChannel(ImageDeliveryChannel imageDeliveryChannel, AssetId assetId)
+    {
+        var presetList = JsonSerializer.Deserialize<List<string>>(imageDeliveryChannel.DeliveryChannelPolicy.PolicyData);
+        List<ObjectInBucket> assetsToDelete;
+        var keys = new List<string>();
+        var containers = new List<string>();
+
+        foreach (var presetIdentifier in presetList)
+        {
+            var presetDictionary = await engineClient.GetAvPresets();
+            
+            if (!presetDictionary.IsNullOrEmpty() && presetDictionary.TryGetValue(presetIdentifier, out var presetName))
+            {
+                var presetDetails = await elasticTranscoderWrapper.GetPresetDetails(presetName);
+
+                if (presetDetails is not null)
+                {
+                    var timebasedFolder = storageKeyGenerator.GetStorageLocationRoot(assetId);
+
+                    var keysFromAws = await bucketReader.GetMatchingKeys(timebasedFolder);
+                    
+                    keys.AddRange(keysFromAws);
+                    containers.Add(presetDetails.Container);
+                }
+            }
+        }
+
+        assetsToDelete = keys.Where(k =>
+                !containers.Contains(k.Split('.').Last()) &&
+                k.Contains(handlerSettings.AssetModifiedSettings.TimebasedKeyIndicator))
+            .Select(k => new ObjectInBucket(handlerSettings.AWS.S3.StorageBucket, k)).ToList();
```

Comment:
I think this method should make use of `TranscoderTemplates`. `.mp4` is a valid extension for both audio and video. This looks like it could lead to accidental deletion of the wrong one.